### PR TITLE
Neo4j init containers

### PR DESCRIPTION
**Pull request by mneedham**

### CODE REVIEW

File: stable/neo4j/templates/core-statefulset.yaml [hunk 1]
```diff
@@ -1,10 +1,17 @@
-apiVersion: "apps/v1beta1"
+apiVersion: "apps/v1beta2"
 kind: StatefulSet
 metadata:
   name: "{{ template "neo4j.core.fullname" . }}"
 spec:
   serviceName: {{ template "neo4j.fullname" . }}
   replicas: {{ .Values.core.numberOfServers }}
+  selector:
+    matchLabels:
```
Comment (unguiculus) [inline stable/neo4j/templates/core-statefulset.yaml:9]: You don't need `chart` and `heritage` here.

File: stable/neo4j/templates/core-statefulset.yaml [hunk 2]
```diff
@@ -1,10 +1,17 @@
-apiVersion: "apps/v1beta1"
+apiVersion: "apps/v1beta2"
```
Comment (unguiculus) [inline stable/neo4j/templates/core-statefulset.yaml]: This probably breaks support for 1.7.

File: stable/neo4j/values.yaml [hunk 3]
```diff
@@ -47,18 +47,34 @@ core:
...
-    #   value: extra-var-value-1
-    # - name: EXTRA_VAR_2
-    #   value: extra-var-value-2
+  ## Pass extra environment variables to the Neo4j container.
+  ##
+  extraVars:
+  - name: EXTRA_VAR_1
+    value: extra-var-value-1
+  - name: EXTRA_VAR_2
+    value: extra-var-value-2
```
Comment (unguiculus) [inline stable/neo4j/values.yaml]: This block should probably stay commented out.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/neo4j/templates/core-statefulset.yaml (hunk 1 above)
Comment (mneedham) [reply stable/neo4j/templates/core-statefulset.yaml:9]: have updated

File: stable/neo4j/values.yaml (hunk 3 above)
Comment (mneedham) [reply stable/neo4j/values.yaml]: yeh good call. Uncommented by accident

**Review by unguiculus — APPROVED** [review]

**Comment by mattfarina** [comment]
@unguiculus here's some output from a failed test run with what's going on with neo4j. https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/charts/2750/pull-charts-e2e/4329/

**Comment by unguiculus** [comment]
@mneedham Could you have a look at the test results, please? Any idea?

**Comment by mneedham** [comment]
@unguiculus I think I've broken something with the selector changes. 

From the logs running this locally:

> 2018-01-08 09:34:34.129+0000 INFO  Resolved initial host 'neo4j-4329-neo4j.default.svc.cluster.local:5000' to []

Let me see what I did wrong...

**Comment by mneedham** [comment]
@unguiculus seems like the issue was using the default namespace everywhere. I've updated everywhere to use the provided namespace.

**Pull request merged**